src/shared/ui/SketchCanvas/SketchCanvas.tsx

```diff
@@ -99,8 +99,17 @@ const SketchCanvas = forwardRef<SketchCanvasRef, Props>((props, ref) => {
   );
 
   const onTouchEnd = useCallback(() => {
+    if (lineSketcher.isDot()) {
+      const firstPoint = lineSketcher.getFirstPoint();
+
+      onTouchProgress({
```

Comment:
I'd suggest creating a separate function that creates a dot. `onTouchProgress` has more responsiblities.